File: flight_delay_prediction/v1/get/ios/FlightDelayPrediction.swift
import Amadeus

var amadeus: Amadeus = Amadeus(
    client_id: "YOUR_API_ID",
    client_secret: "YOUR_API_SECRET"
)

amadeus.travel.predictions.flightDelay.get(data: ["originLocationCode": "NCE",
                                                  "destinationLocationCode": "IST",
                                                  "departureDate": "2020-08-01",
                                                  "departureTime": "18:20:00",
                                                  "arrivalDate": "2020-08-01",
                                                  "arrivalTime": "22:15:00",
                                                  "aircraftCode": "321",
                                                  "carrierCode": "TK",
                                                  "flightNumber": "1816",
                                                  "duration": "PT31H10M"],
                                           onCompletion: {
                                               response, error in
                                               if error == nil {
                                                   print(response!.data)
                                               }
    })

File: flight_delay_prediction/v1/get/Node SDK/flight_delay_prediction.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    // Will there be a delay from BRU to FRA? If so how much delay?
    const response = await amadeus.travel.predictions.flightDelay.get({
      originLocationCode: "NCE",
      destinationLocationCode: "IST",
      departureDate: "2022-08-01",
      departureTime: "18:20:00",
      arrivalDate: "2022-08-01",
      arrivalTime: "22:15:00",
      aircraftCode: "321",
      carrierCode: "TK",
      flightNumber: "1816",
      duration: "PT31H10M",
    });

    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

main();
